lib/dateParser-lambda.js

```diff
@@ -0,0 +1,146 @@
+  }
+  // If timetwister failed to meaningfully do anything, hail mary to at least have a range of years
+  if (Object.values(dateRange).includes(null)) {
+    preparsedDate = parseOnlyYears(preparsedDate)
+  }
+  return dateRange.map((range) => [range.start, range.end])
+}
+
+const timeTwist = async (preparsedDate) => {
+  const payloadStr = JSON.stringify({
+    path: '/',
+    body: JSON.stringify({ dates: [preparsedDate] })
+  })
+  const params = {
+    FunctionName: 'DateParser-production',
+    Payload: payloadStr
+  }
+  try {
+    const { Payload } = await lambda.invoke(params).promise()
+    const payloadParsed = JSON.parse(JSON.parse(Payload).body).dates[preparsedDate]
+    const ranges = payloadParsed.map((result) => ({ start: result.date_start, end: result.date_end }))
+    return ranges
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+// build an object that includes the date as well as classifications based on regex matches
+const classifyDate = (date) => {
+  const classifications = []
+  // vol 7(1992:feb-mar)
+  if (date.match(/\(.*\)/)) classifications.push('dateWithinParens')
+  // 1992:mar-June or 1992:spring
+  if (date.match(/[0-9]{4}:[a-zA-Z]*/)) classifications.push('colon')
+  // 1956/57, but only if it is the only range
+  if (date.match(/^[0-9]{4}\/([0-9]{2}|[0-9]{4})$/)) classifications.push('twoYearRangeSolo')
+  // 1956/57-1981/82 (w 2 or 4 years after slash)
+  if (date.match(/[0-9]{4}\/([0-9]{4}|[0-9]{2})-[0-9]{4}\/([0-9]{4}|[0-9]{2})/)) classifications.push('twoYearRangeMulti')
+  // waiting on timetwister to update to include this
+  // if (date.match(/[a-z]{3}\.?/i)) classifications.push('abbreviatedSeason')
+  // addressing mysterious bug in timetwister that causes null values for XX0X-0X ranges
+  if (date.match(/([0-9]{4})-(0[0-9])/)) classifications.push('0X')
+  // month-month/month ' May-June/July 1963,  Oct. 1961-Sept./Oct. 1962'
+  if (date.match(/((?<=-)[a-z]{3,4}\.?\/|\/[a-z]{3,4}\.|\/(?<=-))/gi)) classifications.push('month-month/month')
+  return classifications
+}
+
+const preparse = (date, classifications) => {
+  let preparsedDate = date
+  classifications.forEach((classification) => {
+    switch (classification) {
+      case 'dateWithinParens':
+        preparsedDate = isolateDateFromWithinParens(preparsedDate)
+        break
+      case 'colon':
+        preparsedDate = removeColonAndFlip(preparsedDate)
+        break
+      case 'twoYearRangeSolo':
+        preparsedDate = soloRangeRemoveSlash(preparsedDate)
+        break
+      case 'twoYearRangeMulti':
+        preparsedDate = twoYearRanges(preparsedDate)
+        break
+      case 'abbreviatedSeason':
+        preparsedDate = fixAbbreviatedSeasons(preparsedDate)
+        break
+      case '0X':
+        preparsedDate = squashShortYearBug(preparsedDate)
+        break
+      case 'month-month/month':
+        preparsedDate = monthRangeWithSlash(preparsedDate)
+        break
+    }
+  })
+  return preparsedDate
+}
```

Comment:
`classifyDate` applies potentially multiple classifications to a single date. But then `preparse` only applies a single relevant transformation. Was it ever your intent to apply multiple transformations - one for each classification for example? Although that could be powerful, I also worry that could be hard to follow. The structure you have here could support applying multiple transformations based on multiple classifications (e.g. by removing some of those `break` lines), but the execution flow is actually more one-to-one (a single transformation is applied when a classification is found). The trouble is it's kind of hard to determine _which_ transformation is going to be applied because the classifications are built in one place, but the transformation that is applied happens here, and you have to read the switch to figure out which transformation is applied because it's based on the order of cases. So given that the structure is in effect already tree like, I wonder if it would be easier to debug if it was made more explicitly so: rather than build a list of `classifications` (suggesting multiple transformations will be applied), apply the "preparsing" at the moment of classification.